.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [macos-latest]
+
+    steps:
+      - name: XCode version
+        run: xcode-select -p
+
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Cache for ccache
+        uses: actions/cache@v1
+        env:
+          cache-name: cache-ccache
+        with:
+          path: ~/.ccache # ccache cache files are stored in `~/.ccache` on Linux/macOS
+          key: ${{ runner.os }}-build-${{ env.cache-name }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
```

Comment:
Incidentally, if you copied this from base it's probably not quite right yet. But with the macOS build constantly breaking itself I haven't really been able to look into it.

Reply:
Yup, totally ripped from base. Remember I have no idea of what I'm doing here :p

Reply:
It's the same general idea as in CircleCI. The cache is a never-changing thing (maybe it's deleted if it's not used for a few months?), so to generate an up to date cache it needs to be sufficiently unique.

https://github.com/koreader/koreader/blob/c20ad8f5e9385dd18138a306b874c5b20302f299/.circleci/config.yml#L19

The problem is that I don't think `${{ runner.os }}-build-${{ env.cache-name }}-` has actually changed since I added the macOS build. So the build time is slowly increasing as the cache becomes increasingly unrelated to the repo.